### ISSUE TITLE
question: The generated xlsx file with a large number of images cannot be opened properly with Excel

### ISSUE THREAD

**Issue by wcshds** [jmcnamara/rust_xlsxwriter#51]
### Question

Hello, I tried to use this crate to generate a basic-full.xlsx file with a lot of images, but the file did not open properly with Excel. The Excel error message is "We found a problem with some content".

Then I tried to generate two files basic-1.xlsx and basic-2.xlsx, each containing half the number of images, which could be opened normally with Excel. I then manually merged the two separately generated files into a single file basic-merge.xlsx, which also opened normally.

I don't know why this crate doesn't generate basic-full.xlsx correctly, can you help me?

[basic-full.xlsx](https://github.com/wcshds/cjk-unicode-pdf-to-xlsx/releases/download/basic/basic-full.xlsx)
[basic-1.xlsx](https://github.com/wcshds/cjk-unicode-pdf-to-xlsx/releases/download/basic/basic-1.xlsx)
[basic-2.xlsx](https://github.com/wcshds/cjk-unicode-pdf-to-xlsx/releases/download/basic/basic-2.xlsx)
[basic-merge.xlsx](https://github.com/wcshds/cjk-unicode-pdf-to-xlsx/releases/download/basic/basic-merge.xlsx)

**Comment by jmcnamara** [comment]
Thanks for the report. I'll take a look.

**Comment by jmcnamara** [comment]
You may already know this but for context a xlsx file is a collection of XML files in a zip container.

I can't find any issue in the XML file structure (although it is hard to be 100% sure since the files are massive).

I think it may be a zipwriter/zip issue. I was able to open the file by unzipping it and then rezipping it. The file is too big to upload here but you can create a rezipped version for testing like this:


```
unzip basic-full.xlsx -d basic-full

cd basic-full
find . -type f | xargs zip ../basic-rezipped.xlsx
cd ..

ls basic-rezipped.xlsx
````

Can you try generate the file like that and then open it. I was able to open it successfully (although it takes a while since the file is huge).

**Comment by wcshds** [comment]
Thanks for your help. After rezipping the file, I can successfully open it now.

**Comment by jmcnamara** [comment]
I think you should submit this as a bug to zip.rs: https://github.com/zip-rs/zip/issues

Attach the links to the basic-full.xlsx and basic-rezipped.xlsx files and add the commands to create one from the other.

Explain that Excel complains about the first file but not when it is rezipped.

**Comment by jmcnamara** [comment]
If you make the rezipped file available I can raise the bug report.

**Comment by wcshds** [comment]
Thanks!
[basic-rezip.xlsx](https://github.com/wcshds/cjk-unicode-pdf-to-xlsx/releases/download/rezip/basic-rezip.xlsx)

**Comment by jmcnamara** [comment]
I've opened an issue in zip.rs for this: https://github.com/zip-rs/zip/issues/388

I tried to do some analysis on the zip file but didn't get anywhere. Hopefully the zip.rs folk(s) will have a look.

I also noticed that there was a u32 overflow issue highlighted by your test program so I put a fix for that in v0.45.0 on crates.io.

**Comment by jmcnamara** [comment]
Closing in favour of https://github.com/zip-rs/zip/issues/388
